src/mui/layout/Sidebar.js

```diff
@@ -6,29 +6,43 @@ import Paper from 'material-ui/Paper';
 import Responsive from './Responsive';
 import { setSidebarVisibility as setSidebarVisibilityAction } from '../../actions';
 
-const styles = {
+
+export function getStyles(props, context) {
+  const {
+    sideBar,
+  } = context.muiTheme;  
+  const styles = {
     sidebarOpen: {
-        flex: '0 0 16em',
+        flex: !(sideBar && sideBar.width) ? '0 0 16em' : `0 0 ${sideBar.width}`,
         marginLeft: 0,
         order: -1,
         transition: 'margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms',
     },
     sidebarClosed: {
-        flex: '0 0 16em',
-        marginLeft: '-16em',
+        flex: !(sideBar && sideBar.width) ? '0 0 16em' : `0 0 ${sideBar.width}`,
+        marginLeft: !(sideBar && sideBar.width) ? '-16em' : `-${sideBar.width}`,
         order: -1,
         transition: 'margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms',
     },
-};
+  };
+
+  return styles;
+}
 
 // We shouldn't need PureComponent here as it's connected
 // but for some reason it keeps rendering even though mapStateToProps returns the same object
 class Sidebar extends PureComponent {
+
+    static contextTypes = {
```

Comment:
Please use the `muiThemeable` HOC (just like `mui/layout/AppBar.js`), I prefer it over grabbing the theme from the context.